src/users/photourl.rs

```diff
@@ -0,0 +1,91 @@
+use std::error::Error;
+
+use url::Url;
+
+#[derive(Debug)]
+pub struct PhotoUrl {
+    url: Url,
+}
+
+impl PhotoUrl {
+    pub fn new(str: &str) -> Result<Self, PhotoUrlError> {
+        let url = Url::parse(str).map_err(PhotoUrlError::InvalidUrl)?;
+
+        if PhotoUrl::has_valid_extension(&url) {
+            Ok(Self { url })
+        } else {
+            Err(PhotoUrlError::InvalidUrlExtension)
+        }
+    }
+
+    pub fn as_url(&self) -> &Url {
+        &self.url
+    }
+
+    fn has_valid_extension(url: &Url) -> bool {
+        let path = url.path();
+        if let Some(filename) = path.split("/").last() {
+            if let Some((name, ext)) = filename.rsplit_once(".") {
+                if !name.is_empty() {
+                    return match ext {
+                        "png" | "jpg" | "jpeg" => true,
```

Comment:
```suggestion
                        "png" | "jpg" | "jpeg" | "webp" => true,
```

Are there any reasons not to include this format?

Reply:
I just hadn't considered it since it's such an annoying format outside the browser.